File: examples/observability-cascade/observability-cascade-node-ruby/api/public.rb
# frozen_string_literal: true

require_relative "../support"
require "holons"
require "relay/v1/relay_pb"

module CascadeNodeRuby
  module Api
    module Public
      class << self
        def tick(request)
          obs = Holons::Observability.current
          slug = responder_slug(obs)
          uid = obs.cfg.instance_uid
          obs.logger("tick").info(
            "tick received",
            "sender" => request.sender,
            "note" => request.note,
            "responder_slug" => slug,
            "responder_uid" => uid
          )
          counter = obs.counter(
            "cascade_ticks_total",
            "Ticks received by this cascade node.",
            "responder_uid" => uid
          )
          counter&.inc
          Relay::V1::TickResponse.new(
            responder_slug: slug,
            responder_instance_uid: uid
          )
        end

        def responder_slug(obs)
          configured = obs.cfg.slug.to_s.strip
          return configured unless configured.empty?

          File.basename($PROGRAM_NAME.to_s).empty? ? "observability-cascade-node-ruby" : File.basename($PROGRAM_NAME.to_s)
        end
      end
    end
  end
end

File: examples/observability-cascade/observability-cascade-node-node/api/public.js
'use strict';

const { observability } = require('@organic-programming/holons');
const pb = require('../gen/node/relay/v1/relay_pb.js');

function tick(request) {
  const obs = observability.current();
  const slug = responderSlug(obs);
  const uid = obs.cfg.instanceUid || '';
  obs.logger('tick').info('tick received', {
    sender: request.getSender(),
    note: request.getNote(),
    responder_slug: slug,
    responder_uid: uid,
  });
  const counter = obs.counter(
    'cascade_ticks_total',
    'Ticks received by this cascade node.',
    { responder_uid: uid },
  );
  if (counter) counter.inc();
  const response = new pb.TickResponse();
  response.setResponderSlug(slug);
  response.setResponderInstanceUid(uid);
  return response;
}

function responderSlug(obs) {
  const configured = String(obs.cfg.slug || '').trim();
  if (configured) return configured;
  return 'observability-cascade-node-node';
}

module.exports = {
  tick,
  responderSlug,
};

